flocker/cli/test/test_deploy_script.py

```diff
@@ -6,33 +6,30 @@
 
 from twisted.python.filepath import FilePath
 from twisted.python.usage import UsageError
-from twisted.trial.unittest import TestCase, SynchronousTestCase
 
 from ...testtools import (
-    FlockerScriptTestsMixin, StandardOptionsTestsMixin)
+    make_flocker_script_test, make_standard_options_test)
 from ..script import DeployScript, DeployOptions
 from ...control.httpapi import REST_API_PORT
 
 
-class FlockerDeployTests(FlockerScriptTestsMixin, TestCase):
+class FlockerDeployTests(
+        make_flocker_script_test(DeployScript, DeployOptions, u'flocker-deploy')
+):
     """Tests for ``flocker-deploy``."""
-    script = DeployScript
-    options = DeployOptions
-    command_name = u'flocker-deploy'
 
 
 CONTROL_HOST = u"192.168.1.1"
 
 
-class DeployOptionsTests(StandardOptionsTestsMixin, SynchronousTestCase):
+class DeployOptionsTests(make_standard_options_test(DeployOptions)):
```

Comment:
:question: The factory returns a class for the standard interface tests. Perhaps the extra tests for the
behaviour of these specific options would be good as a separate class?